### DISCUSSION TITLE
Ambiant dialogue based on HP%, part 2

### DISCUSSION THREAD

**Discussion by Algarad** [Albeoris/Memoria#709] [Q&A] [answered]
Hi all,

I have already created similar post a couple of months ago about this and I got an answer from SamsamTS. With his help, I had managed to have conditions that triggers when a character receives heals when under a specific health percentage, but it is not working as expected anymore, for unknown reason. 

This is an example of the code I am using for Steiner. 
`>Hitted Steiner
VoicePath:Battle/Steiner/StatusRecovery1,StatusRecovery2
[Condition] (EffectTargetFlags & CalcFlag_HpRecovery | EffectTargetFlags & CalcFlag_MpRecovery) && !IsAbilityMultiTarget [/Condition]`

`>Hitted Steiner
VoicePath:Battle/Steiner/HealedEiko1,HealedEiko2
Priority:3
[Condition] EffectTargetFlags & CalcFlag_HpRecovery && CasterCharacterIndex == CharacterId_Eiko && !IsAbilityMultiTarget && (1.0 * HP / MaxHP) <= 0.60 [/Condition]`

What I expect:
- Steiner is healed when he got more than 60% of his health -> StatusRecovery1,StatusRecovery2 files are used
- Steiner is healed when he got less than 60% of his health from Eiko -> HealedEiko1,HealedEiko2 files are used

What happens:
- StatusRecovery1/2 files are reproduced even if Steiner has less health than 60%
- I have the feeling HealedEiko1/2 are reproduced only when Steiner has about 40% or lesser, while the quota should be 60%.

Can someone help me with this? I get the feeling that the calculations are not working properly, which sounds stupid...

Thank you!
Algarad

**Comment by Tirlititi** [comment, marked as answer]
Hi, that's because the trigger happens right after the damage / heal is done, so it depends on whether Steiner has 60% of his max HP after the heal there.

You can use that formula for the heal check instead:
```txt
HP - HPDamage <= 0.6 * MaxHP
```
Hopefully that fixes it.
Keep it up!

**Comment by Algarad** [comment]
Hi,
thank you for your explanation. Now I also understand better the original formula Samsam gave me.
So, my condition was only activated when a character received healings and his HP was less than 60% **after the healing**.

I already did some test with your formula and it seems like it's working fine.
Thank you again!
